src/components/visualizations/chart/highcharts/plugins/zeroAlignYAxis.ts

```diff
@@ -246,56 +246,60 @@ export function getTickAmount(tickPosition: number[]): number {
-    let minmax: IMinMaxInfo[];
+export function getTickPositioner() {
+    // persist tick amount on the axis has larger value, then apply it to opposite axis
+    let tickAmount: number = 0;
+    const minmax: IMinMaxInfo[] = [];
 
     return function(min: number, max: number) {
+        const currentAxisIndex = this.options.index;
         if (typeof min === 'undefined' || min === null || typeof max === 'undefined' || max === null) {
-            return;
+            // mark this axis invalid
+            minmax[currentAxisIndex] = {
+                id: currentAxisIndex,
+                min: null,
+                max: null,
+                isSetMin: this.userOptions.min !== undefined,
+                isSetMax: this.userOptions.max !== undefined
+            };
+            return [];
         }
 
         const dataMin = Math.min(0, this.dataMin, this.dataMax);
         const dataMax = Math.max(0, this.dataMin, this.dataMax);
-        if (min === max || min >= dataMax || max <= dataMin) {
-            const tickNum = this.tickPositions.length;
-            if (tickNum === 1 && this.tickPositions[0] === 0) {
-                // nothing to render on this axis
-                return [];
-            }
-            return tickNum ? createArrayFromRange(min, max, this.tickAmount) : this.tickPositions;
+        // indicate pair of min/max is out of range of data
+        if (min >= max || min >= dataMax || max <= dataMin) {
+            // nothing to render on this axis, so that hide it
+            return [];
         }
 
         const chart = this.chart;
-
-        if (!minmax) {
-            // ----------------------------
-            // Computed once for both axes
-            // ----------------------------
-            minmax = [];
-
-            for (const axis of chart.axes) {
-                if (axis.coll === 'yAxis') {
-                    if (typeof axis.max === 'undefined' || axis.max === null ||
-                        typeof axis.min === 'undefined' || axis.min === null) {
-                        // Don't have min/max values for both axes yet.
-                        // Clean up processed values from other axis.
-                        minmax = undefined;
-                        // Exit. Process in next call.
-                        return createArrayFromRange(min, max, this.tickAmount);
-                    }
-                    minmax.push({
-                        id: axis.options.index,
-                        min: axis.min,
-                        max: axis.max,
-                        isSetMin: axis.userOptions.min !== undefined,
-                        isSetMax: axis.userOptions.max !== undefined
-                    });
+        const yAxes = chart.axes.filter((axis: any) => axis.coll === 'yAxis');
+
+        for (const yAxisIndex in yAxes) {
+            if (yAxes.hasOwnProperty(yAxisIndex)) {
```

Comment:
this processing of other axes can be moved to separate function

Reply:
Actually, `for...in` and `hasOwnProperty` must be a pair to bypass typescript check.
It does not really relate to process opposite axis.